corehq/apps/groups/models.py

```diff
@@ -222,6 +222,13 @@ def by_user(cls, user_or_user_id, wrap=True, include_names=False):
             return [r['id'] for r in results]
 
     @classmethod
+    def get_case_sharing_accessible_locations(cls, domain, user):
+        return [
+            location.case_sharing_group_object() for location in
+            SQLLocation.objects.accessible_to_user(domain, user).filter(location_type__shares_cases=True)
+        ]
```

Comment:
You could also call this method below in `get_case_sharing_groups` to avoid duplication